src/lib/src/grid3d/grid.cpp

```diff
@@ -421,4 +421,146 @@ get_bounding_box(const Grid &grid)
     return std::make_tuple(minv, maxv);
 }
 
+inline double
+interp1d(const double x1, const double x2, const double fx)
+{
+    return x1 + fx * (x2 - x1);
+};
+
+inline xyz::Point
+interp(const xyz::Point pt1, const xyz::Point pt2, const double f)
+{
+    return xyz::Point{ pt1.x + f * (pt2.x - pt1.x), pt1.y + f * (pt2.y - pt1.y),
+                       pt1.z + f * (pt2.z - pt1.z) };
+}
```

Comment:
There is already a function for this, "lerp3d" in [numerics](https://github.com/equinor/xtgeo/blob/main/src/lib/include/xtgeo/numerics.hpp), although perhaps even better to have in `geometry_basics.hpp`. Same with `interp1d`; move tto `geometry_basics.hpp`